app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb

```diff
@@ -34,7 +34,37 @@ def parse_ssps
         :name        => ssp.name,
         :total_space => ssp.capacity.to_f.gigabytes.round, # hmc returns a str in byte
         :ems_ref     => ssp.uuid,
-        :free_space  => ssp.free_space.to_f.gigabytes.round
+        :free_space  => ssp.free_space.to_f.gigabytes.round 
+      )
+    end
+  end
+
+  def parse_vm_disks(lpar, hardware)
+    lpar_lu_mappings = collector.vioses.flat_map do |vios|
+      vios.vscsi_mappings.select do |mapping|
+        mapping.lpar_uuid == lpar.uuid && mapping.storage.kind_of?(IbmPowerHmc::LogicalUnit)
+      end
+    end
```

Comment:
Best to cache this in your collector so that we don't have to re-run this loop for every disk.

```ruby
def vscsi_luns
  @vscsi_luns ||= vioses.flat_map { |vios| vios.vscsi_mappings.select { |mapping| mapping.storage.kind_of?(IbmPowerHmc::LogicalUnit) } }
end

def vscsi_luns_by_uuid
  @vscsi_luns_by_uuid ||= vscsi_luns.group_by(&:lpar_uuid)
```

Reply:
Thanks @agrare for your review ! I'm back. 
It looks like `collector.vscsi_lun_mappings[lpar.uuid]` is waiting for an integer as a key. I think there is another way to select the right element ?

Reply:
Finally working : 
```
def parse_vm_disks(lpar, hardware)
     collector.vscsi_lun_mappings_by_uuid[lpar.uuid].to_a.each do |mapping|
        ...
    end
```